### ISSUE TITLE
[Bug]: No answer from KI in Assistent

### ISSUE THREAD

**Issue by cheesynator** [nextcloud/assistant#142]
### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Here is the translation:

"I'm running Nextcloud on my Unraid server, which I have set up in a Docker container using the Linuxserver Image. Additionally, I've installed a local AI (Ollama and LocalAI) as well.

When I enter the IP address of the AI (Ollama or LocalAI) into the App LocalAI under Nextcloud, the AI is found and displays the installed LLMs. The LLMs are running smoothly and I can chat with the KI using OpenWebUI or through the chat in (LocalAI).

However, when I start a chat in the Nextcloud Assistant, I don't receive an answer. However, my question goes into Ollama and LocalAI (which can be seen in their logs) but is not processed.

But if I set up a prompt using the Nextcloud App on my iPhone under "More / Assistant", I immediately get a response from the KI.

After long searching, I was able to pinpoint the error. Under Nextcloud 29.0.7 (Docker Nextcloud:29.0.7) the response works fine. But with Nextcloud 30, I don't receive any answer. 

"I followed your test environment.

1. I installed the two Nextcloud Docker images:
   docker pull nextcloud:latest` and `nextcloud:29.0.7` (the official images)
2. I created an admin account on the welcome screen, using SQLite as the database.
3. I then installed the Assistent and LocalAI apps under Nextcloud 29.0.7 and 30 in the "Apps" section.
4. Under Settings > Artificial Intelligence > Admin area, I entered the IP address of the local AI (LocalAI and then Ollama).
5. The LLMs were displayed. I tested with llama3.2 and gpt4.
6. I opened the Assistent and started a chat with the KI.
7. Under 29.0.7, I received an immediate answer. Under 30, no answer was returned.
8. However, the KI responds to both 29.0.7 and 30 when using the Nextcloud iPhone app.

Note: I didn't make any changes or configurations during these tests. The first start only created the admin user, and both tests were run with minimal system setups."

I hope you can help me find a solution to this problem."

Greetings Thomas

### Steps to reproduce

1.
2.
3.

### Expected behavior

.

### Nextcloud Server version

30

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by ascheurer-rts** [comment]
@joshtrichards not sure, if "Assistant" is the rights section, because tho code change needs to be done in server, not in Assistant.
And the issue effects not only the Assistant, but potential all app that are using "worker-src"

**Comment by jasoncheng7115** [comment]
Yes, I was using NC 29 with LM Studio and it worked perfectly, but after I upgraded to NC 30, the Assistant stopped responding.

**Comment by hdnh2006** [comment]
> As mentioned by [@ascheurer-rts](https://github.com/ascheurer-rts)
> 
> This is because since NC 30, tasks are running in a background job. So there is a delay, you wait until the next cron.php execution. This delay can be avoided by running an occ command that executes a specific background job type ASAP. More info at https://docs.nextcloud.com/server/latest/admin_manual/ai/overview.html#ai-overview-improve-ai-task-pickup-speed


But Julien, the tasks doesn't finish, that's what we are trying to explain. It is not related with a late response, it is related with no action is taken from NC, can you please try to reproduce the issue?

**Comment by MarslMarcello** [comment]
Hi, might it be there is an issue/bug with the config at all in NC 30.x for the A.I. Assistant?
I have two NC instances which are using the same Ollama Server. NC 29 works w/o any issue and the config is I just dropped the BASE URL to my server w/o anything behind like /v1 and models are listed and show in the config in NC.
 
My main NC version 30.0.2 has the same issue what the thread creator is describing no response. 
The following things I found so far. Within my config for A.I. I have to tell now baseURL.de/v1 and see an 
{"OK (| 200 |    2.208078ms |  IP.xxx.xxx.xxx.xxx | GET      "/v1/models"_ } 
in the Ollama logs + to get not the error, no models found. 
When I trigger the assistant, I'm not getting any responses and can see in the NC logs the following error:
{GET https://baseURL.de/v1/v1/models` resulted in a `404 Not Found` response: 404 page not found}.
So something seems to be wrong.

**Comment by TheRsKing** [comment]
> Had the same problem but but in a LAMP stack installation. Try following: Exec the occ command inside the container `occ background-job:worker` while this commend is running, start a AI conversation. If you get an answer, the problem is the not running background job, or the job is scheduled and not running permanent

i can confirm

**Comment by janepie** [comment]
I'm closing this as it's a duplicate of #10
